firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalSerializer.java

```diff
@@ -48,30 +45,27 @@ public LocalSerializer(RemoteSerializer rpcSerializer) {
   }
 
   /** Encodes a MaybeDocument model to the equivalent protocol buffer for local storage. */
-  com.google.firebase.firestore.proto.MaybeDocument encodeMaybeDocument(MaybeDocument document) {
+  com.google.firebase.firestore.proto.MaybeDocument encodeMaybeDocument(MutableDocument document) {
     com.google.firebase.firestore.proto.MaybeDocument.Builder builder =
         com.google.firebase.firestore.proto.MaybeDocument.newBuilder();
 
-    if (document instanceof NoDocument) {
-      NoDocument noDocument = (NoDocument) document;
-      builder.setNoDocument(encodeNoDocument(noDocument));
-      builder.setHasCommittedMutations(noDocument.hasCommittedMutations());
-    } else if (document instanceof Document) {
-      Document existingDocument = (Document) document;
-      builder.setDocument(encodeDocument(existingDocument));
-      builder.setHasCommittedMutations(existingDocument.hasCommittedMutations());
-    } else if (document instanceof UnknownDocument) {
-      builder.setUnknownDocument(encodeUnknownDocument((UnknownDocument) document));
-      builder.setHasCommittedMutations(true);
+    if (document.isNoDocument()) {
+      builder.setNoDocument(encodeNoDocument(document));
+    } else if (document.isFoundDocument()) {
+      builder.setDocument(encodeDocument(document));
+    } else if (document.isUnknownDocument()) {
+      builder.setUnknownDocument(encodeUnknownDocument(document));
     } else {
-      throw fail("Unknown document type %s", document.getClass().getCanonicalName());
+      throw fail("Unknown Document %s", document);
```

Comment:
Consider rewording this error message since the concept of an "unknown document" is already taken, and would have caused the "else if" branch above to be entered had the document been "unknown".